go-controller/pkg/ovn/gatewayrouter/policybasedroutes.go

```diff
@@ -0,0 +1,209 @@
+package gatewayrouter
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	utilnet "k8s.io/utils/net"
+
+	"github.com/ovn-org/libovsdb/client"
+
+	libovsdbops "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/ops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+type PolicyBasedRoutesManager struct {
+	clusterRouterName string
+	netInfo           util.NetInfo
+	nbClient          client.Client
+}
+
+func NewPolicyBasedRoutesManager(nbClient client.Client, clusterRouterName string, netInfo util.NetInfo) *PolicyBasedRoutesManager {
+	return &PolicyBasedRoutesManager{
+		clusterRouterName: clusterRouterName,
+		netInfo:           netInfo,
+		nbClient:          nbClient,
+	}
+}
+
+func (pbr *PolicyBasedRoutesManager) Add(nodeName, mgmtPortIP string, hostIfCIDR *net.IPNet, otherHostAddrs []string) error {
```

Comment:
nit: maybe a bit defensive and i know the callers are doing everything right but id still love some validation of the `hostIfCIDR` to ensure its not nil before accessing its data.

Reply:
Done